lib/member_holding_header_factory.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require "custom_errors"
+require "member_holding_header"
+
+# Usage:
+#
+# MemberHoldingHeaderFactory.new("mono", header_str)
+#   .get_instance # -> MonoHoldingHeader
+#
+# MemberHoldingHeaderFactory.new("multi", header_str)
+#   .get_instance # -> MultiHoldingHeader
+#
+# MemberHoldingHeaderFactory.new("serial", header_str)
+#   .get_instance # -> SerialHoldingHeader
+class MemberHoldingHeaderFactory
+  MON = "mono"
+  MUL = "multi"
+  SER = "serial"
+
+  def initialize(item_type, header_line)
+    @item_type   = item_type
+    @header_line = header_line
+  end
+
+  # Return a proper subclass of MemberHoldingHeader
+  # rubocop:disable Naming/AccessorMethodName
+  def get_instance
```

Comment:
We have a couple different styles of factories going on here; I'd generally lean towards preferring the `Whatever.for(something)` style, and I think it's preferable to be consistent if possible

Reply:
Agree. Fixed.